enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -321,32 +324,48 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
             3-tuple of (bool, str, list of dict):
                 * first element is true if the learner can redeem the content,
                 * second element contains a reason code if the content is not redeemable,
-                * third a list of any transactions represending existing redemptions (any state).
+                * third a list of any transactions representing existing redemptions (any state).
         """
+        content_metadata = self.get_content_metadata(content_key)
+        subsidy_can_redeem_payload = self.subsidy_client.can_redeem(
+            self.subsidy_uuid,
+            lms_user_id,
+            content_key,
+        )
+
+        active_subsidy = subsidy_can_redeem_payload.get('active', False)
+        existing_transactions = subsidy_can_redeem_payload.get('all_transactions', [])
+
+        # inactive subsidy
+        if not active_subsidy:
+            return (False, REASON_SUBSIDY_EXPIRED, [])
+
+        # inactive policy with spend remaining (could have redeemed)
+        if not self.active and not self.will_exceed_spend_limit(content_key, content_metadata=content_metadata):
```

Comment:
This and the changes on like 149 are a little sus i think. I synced with @brobro10000 on slack with some suggestions. He's going to take another look at the logic in here.

Reply:
Going to take another look at it after our discussion yesterday 👍🏽